pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -778,114 +781,191 @@ func TestProgressingCondition(t *testing.T) {
-		workloadcontroller.EnsureAtMostOnePodPerNode,
-		recorder,
-		operatorConfig,
-		"my.co/repo/img:latest",
-		operatorConfig.Status.Generations,
-		proxyLister,
-		specAnnotations,
-	)
+			recorder := events.NewInMemoryRecorder("", clock.RealClock{})
 
-	if err != nil {
-		t.Fatalf("unexpected error: %s", err.Error())
-	}
-	if !rcBool {
-		t.Fatal("apply deployment does not think a changes was made")
-	}
+			operatorConfig := &operatorv1.OpenShiftControllerManager{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "cluster",
+					Generation: 2,
+				},
+				Spec: operatorv1.OpenShiftControllerManagerSpec{
+					OperatorSpec: operatorv1.OperatorSpec{},
+				},
+				Status: operatorv1.OpenShiftControllerManagerStatus{
+					OperatorStatus: operatorv1.OperatorStatus{
+						ObservedGeneration: 2,
+					},
+				},
+			}
 
-	if ds == nil {
-		t.Fatalf("nil deployment returned")
-	}
+			specAnnotations := map[string]string{
+				"openshiftcontrollermanagers.operator.openshift.io/cluster": strconv.FormatInt(operatorConfig.ObjectMeta.Generation, 10),
+				"configmaps/config":               "54587",
+				"configmaps/client-ca":            "12515",
+				"configmaps/openshift-service-ca": "45789",
+				"configmaps/openshift-global-ca":  "56784",
+			}
 
-	foundNoProxy := false
-	foundHTTPProxy := false
-	foundHTTPSProxy := false
-	for _, c := range ds.Spec.Template.Spec.Containers {
-		for _, e := range c.Env {
-			switch e.Name {
-			case "NO_PROXY":
-				if e.Value == proxyConfig.Status.NoProxy {
-					foundNoProxy = true
-				}
-			case "HTTP_PROXY":
-				if e.Value == proxyConfig.Status.HTTPProxy {
-					foundHTTPProxy = true
-				}
-			case "HTTPS_PROXY":
-				if e.Value == proxyConfig.Status.HTTPSProxy {
-					foundHTTPSProxy = true
-				}
+			countNodes := func(nodeSelector map[string]string) (*int32, error) {
+				result := int32(3)
+				return &result, nil
 			}
-		}
-	}
 
-	if !foundNoProxy {
-		t.Fatalf("NO_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
-	}
-	if !foundHTTPProxy {
-		t.Fatalf("HTTP_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
-	}
-	if !foundHTTPSProxy {
-		t.Fatalf("HTTPS_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
+			// Sync
+			ds, rcBool, err := manageOpenShiftControllerManagerDeployment_v311_00_to_latest(
+				tt.mustLoadAsset,
+				deployClient,
+				countNodes,
+				workloadcontroller.EnsureAtMostOnePodPerNode,
+				recorder,
+				operatorConfig,
+				"my.co/repo/img:latest",
+				operatorConfig.Status.Generations,
+				proxyLister,
+				specAnnotations,
+			)
+
+			if err != nil {
+				t.Fatalf("unexpected error: %s", err.Error())
+			}
+			if !rcBool {
+				t.Fatal("apply deployment does not think a changes was made")
+			}
+			if ds == nil {
+				t.Fatalf("nil deployment returned")
+			}
+
+			// Check the resulting env
+			for _, c := range ds.Spec.Template.Spec.Containers {
+				if !cmp.Equal(c.Env, tt.expectedEnv) {
+					t.Errorf(cmp.Diff(tt.expectedEnv, c.Env))
```

Comment:
nit: we could make the error message a little bit more verbose

Reply:
Amended.